Microsoft.SystemForCrossDomainIdentityManagement/Protocol/ProtocolExtensions.cs

```diff
@@ -148,9 +148,14 @@ private static void Apply(this Core2Group group, PatchOperation2 operation)
                                                  Value = item.Value
                                              })
                                      .ToArray();
-                                group.Members =
-                                    group.Members != null ?
-                                        group.Members.Concat(newMembers).ToArray() : newMembers;
+
+                                foreach(Member member in newMembers)
+                                {
+                                    if (!group.Members.Any(M => M.Value == member.Value))
```

Comment:
Format LINQ statements in the following way.
!group.Members.Any((member item) => string.Equals(item, member.Value, StringComparison.OrdinalIgnoreCase)

StringComparison instead of "==" because the SCIM client may or may not send the member id in the same case. If the SCIM server requires the string comparison to be case sensitive, then they may do so.

Another way would be to expose an Equals method on the Members and then compare the members for equality, where the Server decides how it wants to implement Equals.